### PR TITLE
linux: add 4.5

### PR DESCRIPTION
**Pull request by kragniz**
My poor laptop is still building this locally. I'll update this pr once it's been tested.

### CODE REVIEW

File: pkgs/os-specific/linux/kernel/linux-4.5.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{ stdenv, fetchurl, perl, buildLinux, ... } @ args:
+
+import ./generic.nix (args // rec {
+  version = "4.5";
+  extraMeta.branch = "4.5";
+
```
Comment (NeQuissimus) [inline pkgs/os-specific/linux/kernel/linux-4.5.nix]: I think you will need `modDirVersion = 4.5.0`.

See #12313

Comment (zimbatm) [inline pkgs/os-specific/linux/kernel/linux-4.5.nix]: Yep, got the issue when building this branch

Comment (kragniz) [inline pkgs/os-specific/linux/kernel/linux-4.5.nix]: Yup, mine failed also. I've updated the patch and kicked off another build.

**Comment by NeQuissimus** [comment]
This should be good to merge now. I will install it at home once it is in master.

**Comment by kragniz** [comment]
Built locally.

**Pull request merged**